sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/job/SubscriptionScheduler.java

```diff
@@ -0,0 +1,40 @@
+package com.exadel.sandbox.team5.service.job;
+
+import com.exadel.sandbox.team5.dao.EmployeeDAO;
+import com.exadel.sandbox.team5.dao.ParamsDAO;
+import com.exadel.sandbox.team5.service.MailSenderService;
+import com.exadel.sandbox.team5.util.Pair;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.MultimapBuilder;
+import com.google.common.collect.Multimaps;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.Instant;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Slf4j
+@Transactional
+@Service
+@Component
+public class SubscriptionScheduler {
+    private final EmployeeDAO employeeDAO;
+    private final ParamsDAO paramsDAO;
+    private final MailSenderService mailSender;
+
+    private static final int delay = 3600000 * 12;
+
+    @Scheduled(fixedRate = delay)
+    public void senderSchedule() {
+        Multimap<String, String> emails = employeeDAO.getNewDiscounts().stream()
```

Comment:
method `getNewDiscounts` method contains the following condition `d.periodStart > p.value`. It means that if discount starts after more that one day then users will receive a message about this discount several times. I think we should send message about new created discount only one time for each subscribed user

Reply:
it send discount only if the periodStart of discount is after last mailing.

Reply:
done